src/api/main.py

```
@@ -0,0 +1,285 @@
+if USER_DB_PATH.exists():
+    try:
+        with open(USER_DB_PATH, "r") as f:
+            USER_DB = json.load(f)
+            logger.info(f"Loaded {len(USER_DB)} user(s) from credentials file")
+    except Exception as e:
+        logger.error(f"Error loading credentials file: {e}")
+        # Create a backup of the corrupted file if it exists
+        if os.path.getsize(USER_DB_PATH) > 0:
+            backup_path = USER_DB_PATH.with_suffix(".json.bak")
+            try:
+                shutil.copy(USER_DB_PATH, backup_path)
```

Ruff (0.11.9)

Undefined name `shutil`

(F821)